### PR TITLE
feat(customers): Find customers by email

### PR DESCRIPTION
**Pull request by willbarre2**
As described here
https://stripe.com/docs/api/customers/list#list_customers-email,
the Stripe API allows you to find one or more customers by their
email addresses by adding the query param `?email=<something>`
to the `/v1/customers` route.

Stripe doesn't care about the format of the emails we send in
the request, as long as they are strings.
If we request with an unknown address, it returns an empty list.
On the other hand, if we request with a known address, it returns
a list of customers who have this address in their email field.
So there can be several customers with the same email address.

Let's emulate this behavior.

Closes: https://github.com/adrienverge/localstripe/pull/154 https://github.com/adrienverge/localstripe/issues/153

### CODE REVIEW

File: localstripe/resources.py
Comment (bravier) [inline localstripe/resources.py:821]: Can you put this function lower next to other `_api_list*` functions?

File: localstripe/resources.py
Comment (bravier) [inline localstripe/resources.py:829]: We usually put this check first in the functions (see other places in the code).

File: test.sh
Comment (bravier) [inline test.sh:25]: Can you grep the whole line like at other places? (with the comma at the end)

File: test.sh
Comment (bravier) [inline test.sh:24]: This can fit on the previous line.

**Review by bravier — COMMENTED** [review]
Hello @willbarre2,
Thanks for your first contribution on Localstripe ;)
A few comments!

**Review by adrienverge — COMMENTED** [review]
Thanks for your contribution @willbarre2!

Nothing to add to @bravier's review (thank you by the way ;))

File: localstripe/resources.py
Comment (willbarre2) [reply localstripe/resources.py:821]: Yes, of course.

File: test.sh
Comment (willbarre2) [reply test.sh:25]: OK, so the commas are also missing in the lines 795 800 805 926 935, in the `grep -oE '"total_count":...`

File: localstripe/resources.py
Comment (willbarre2) [reply localstripe/resources.py:829]: Ok

File: test.sh
Comment (bravier) [reply test.sh:25]: That's true, we missed it during previous reviews!
Feel free to open another PR to fix that if you want ;)

**Comment by willbarre2** [comment]
A new push is ready to be reviewed.

- `_api_list_all()` moved
- kwargs check moved
- whole line (+ comma) greped
- line 23 and 24 (`test.sh` file) merged

File: localstripe/resources.py
Comment (bravier) [inline localstripe/resources.py:971]: Can you put an empty line before this one please?

**Review by bravier — COMMENTED** [review]
Looks good to me too!
Only a few minor comments to address before merge.

File: localstripe/resources.py
Comment (willbarre2) [reply localstripe/resources.py:979]: OK

File: localstripe/resources.py
Comment (willbarre2) [reply localstripe/resources.py:972]: Ok

**Comment by willbarre2** [comment]
A new push is ready to be reviewed.
- extra line removed
- empty line added

**Review by adrienverge — APPROVED** [review]
I let @bravier have the final word, but this last version looks perfect to me!

**Review by bravier — APPROVED** [review]
LGTM!

**Pull request merged**
